streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableSourceTest.java

```diff
@@ -156,7 +156,7 @@ public void kTableShouldLogAndMeterOnSkippedRecords() {
                     .filter(e -> e.getLevel().equals("WARN"))
                     .map(Event::getMessage)
                     .collect(Collectors.toList()),
-                hasItem("Skipping record due to null key. topic=[topic] partition=[0] offset=[0]")
+                hasItem("Skipping record due to null key. value=[value] topic=[topic] partition=[0] offset=[0]")
```

Comment:
I probably don't need to point this out, but this will have to change back when you remove the value from the production code.

On another note, I guess we could add a test for the other (new) code path when the metadata is absent. I'll leave it up to you.

Reply:
> On another note, I guess we could add a test for the other (new) code path when the metadata is absent. I'll leave it up to you.

Happy to add more tests. What I'm wondering is how to test scenarios where no context is added 🤔 ? TopologyTestDriver allows to nullify this?